### PR TITLE
User model avatars

### PR DESCRIPTION
**Pull request by codev**
I didn't want to use gravatar on a recent project for anonymity but wanted avatars from my current user class. So I've added the ability to set Forem.avatar_user_method to a method to be called on user to return an image url to use as an avatar. If you don't set avatar_user_method you keep the gravatar functionality.

### CODE REVIEW

**Comment by parndt** [comment]
Odd, it's saying there are merge conflicts with this pull request. Can you please rebase against master?

Thanks!

**Comment by codev** [comment]
Any better?

**Comment by parndt** [comment]
Yeah, it now is clean.

**Comment by parndt** [comment]
Or was, sorry.

**Comment by codev** [comment]
Merged it with all the moderation fixes

**Comment by radar** [comment]
Please extract changes pertinent to this pull request into another branch and re-submit a pull request. There are far too many commits in here with too many changes for my liking.

For instance, you have a commit that removes subscriber functionality and `lib/forem.rb` removes the `forem_autocomplete` attribute.

File: spec/dummy/db/migrate/20120208202625_add_custom_avatar_url_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddCustomAvatarUrlToUsers < ActiveRecord::Migration
```
Comment (radar) [inline spec/dummy/db/migrate/20120208202625_add_custom_avatar_url_to_users.rb:1]: spec/dummy is re-generated with a call to `rake forem:dummy_app`, and so this migration would be nuked by that call. Recommended to move it into the dummy generator.

File: app/helpers/forem/posts_helper.rb [hunk 2]
```diff
@@ -1,8 +1,16 @@
 module Forem
   module PostsHelper
     def avatar(user, options = {})
-      if (email = user.try(:email)).present?
-        image_tag(avatar_url(email, options), :alt => "Gravatar")
+      if Forem.avatar_user_method.nil?
```
Comment (parndt) [inline app/helpers/forem/posts_helper.rb:4]: if `avatar_user_method` just returned `:email` by default then you wouldn't need the conditional logic here at all :-)

Comment (parndt) [inline app/helpers/forem/posts_helper.rb:4]: By this, I mean that `avatar_url` would need to be modified to handle all cases automatically.

**Pull request closed without merging**